src/openMVG/voctree/database.cpp

```
@@ -169,83 +172,258 @@ void Database::loadWeights(const std::string& file)
+      {
+        N1 += i1->second.size()*word_weights_[i1->first];
+         ++i1;
+      }
+        if( ( fabs(i1->second.size() - 1.0) < epsilon ) && ( fabs(i2->second.size() - 1.0) < epsilon) )
+        {
+          score += word_weights_[i1->first];
+          N1 += word_weights_[i1->first];
+          N2 += word_weights_[i2->first];
+        }
+        ++i1;
+        ++i2;
+    }
+
+    while(i1 != i1e)
+    {
+      N1 += i1->second.size()*word_weights_[i1->first];
+      ++i1;
+    }
+
+    while(i2 != i2e)
+    {
+      N2 += i2->second.size()*word_weights_[i2->first];
+      ++i2;
+    }
+    
+    distance = - score;
   }
+  
+  else if(distanceMethod.compare("inversedWeightedCommonPoints") == 0)
+  {
+    double score = 0.0;
+    double N1 = 0.0;
+    double N2 = 0.0;
+    std::map<int,int> compteur;
+    
+    while(i1 != i1e && i2 != i2e)
+    {
+      if(i2->first < i1->first)
+      {
+        N2 += i2->second.size() / word_weights_[i2->first];
+        ++i2;
+      }
+      else if(i1->first < i2->first)
+      {
+        N1 += i1->second.size() / word_weights_[i1->first];
+         ++i1;
+      }
+      else
+      {
+        compteur[i1->first] += std::min(i1->second.size(), i2->second.size());
+        N1 += i1->second.size() / word_weights_[i1->first];
+        N2 += i2->second.size() / word_weights_[i2->first];
+        ++i1;
+        ++i2;
+      }
+    }
+
+    while(i1 != i1e)
+    {
+      N1 += i1->second.size() / word_weights_[i1->first];
+      ++i1;
+    }
 
-  while(i2 != i2e)
+    while(i2 != i2e)
+    {
+      N2 += i2->second.size() / word_weights_[i2->first];;
+      ++i2;
+    }
+    
+    for(auto iCompteur = compteur.begin(); iCompteur != compteur.end(); iCompteur++)
+      score += (1.0/iCompteur->second) * word_weights_[iCompteur->first];
+    
+    distance = - score;
+  }
+  else
   {
-    distance += i2->second;
-    ++i2;
+    std::cout << "Unknown distanceMethod" << std::endl;
+    exit(-1);
```

@jmelou @fabiencastan I'd rather throw an exception, something like
`throw std::invalid_argument("distance method "+method+" unknown!")`